src/dali_backend.cc

```diff
@@ -220,6 +220,10 @@ class DaliModelInstance : public ::triton::backend::BackendModelInstance {
     return ret;
   }
 
+  int32_t DetermineDeviceId() {
+    return !CudaStream() ? -1 : device_id_;
```

Comment:
What is the value of `device_id_` when `!CudaStream()` ?

Reply:
There's no clear explanation for this in Triton docs (the `device_id_` in this case comes from `backend_model_instance.h`). When I logged it, it was `0`, so I suppose we can't trust this. However, from Triton's docs:
```c++
  // Returns the stream associated with this instance that can be used
  // for GPU<->CPU memory transfers. Returns nullptr if GPU support is
  // disabled or if this instance is not executing on a GPU.
  cudaStream_t CudaStream() { return stream_; }
```
Thus `CudaStream` seems it can be trusted in determining, whether the GPU is available

Reply:
Understood